packages/webapp/src/containers/Log/HarvestUseType/index.js

```diff
@@ -26,40 +26,57 @@ function HarvestUseType() {
   useEffect(() => {}, []);
 
   const onBack = (data) => {
+    dispatch(canEditStepTwo(false));
     const tempProps = JSON.parse(JSON.stringify(data));
+
     if (defaultData.selectedUseTypes.length > 0) {
-      tempProps.selectedUseTypes.map((item, idx) => {
-        if (
-          idx < defaultData.selectedUseTypes.length &&
-          item.harvest_use_type_name === defaultData.selectedUseTypes[idx].harvest_use_type_name
-        ) {
-          item.quantity_kg = defaultData.selectedUseTypes[idx].quantity_kg;
-        }
+      tempProps.selectedUseTypes.map((item) => {
+        defaultData.selectedUseTypes.map((item1) => {
+          let name = isEditStepTwo.isEditStepTwo
+            ? item1.harvestUseType.harvest_use_type_name
+            : item1.harvest_use_type_name;
+          if (item.harvest_use_type_name === name) {
+            item.quantity_kg = item1.quantity_kg;
+          }
+        });
       });
     }
     defaultData.selectedUseTypes = tempProps.selectedUseTypes;
```

Comment:
Can you change this? since defaultData is retrieved from state, we should not be mutating it it should always be changed to a new value. instead in the next line that dispatches. do 
```
dispatch(harvestLogData( {...defaultData, selectedUseTypes: (newData or tempProps.selectedUseTypes)} ))
```